compiler/src/dmd/typinf.d

```diff
@@ -47,6 +47,7 @@ extern (C++) void genTypeInfo(Expression e, const ref Loc loc, Type torig, Scope
     {
         if (!global.params.useTypeInfo)
         {
+            global.gag = 0;
```

Comment:
You have the same but 11 lines below

Reply:
There's other places that use `fatal`. I considered doing this for all call sites or better, update fatal to receive an error message and just call it with whatever is passed to error. However, these should be done in later PR.